src/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -1209,34 +1235,48 @@ public void Range_Add_Success(Uri remoteServer)
-        public async Task GetResponseAsync_PostRequestStream_ContainsData(Uri remoteServer)
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task GetResponseAsync_PostRequestStream_ContainsData(bool useSsl)
         {
-            HttpWebRequest request = HttpWebRequest.CreateHttp(remoteServer);
-            request.Method = HttpMethod.Post.Method;
+            var options = new LoopbackServer.Options { UseSsl = useSsl };
 
-            using (Stream requestStream = await request.GetRequestStreamAsync())
+            await LoopbackServer.CreateClientAndServerAsync(async uri =>
             {
-                requestStream.Write(_requestBodyBytes, 0, _requestBodyBytes.Length);
-            }
+                HttpWebRequest request = WebRequest.CreateHttp(uri);
+                request.ServerCertificateValidationCallback = (a, b, c, d) => true;
+                request.Method = HttpMethod.Post.Method;
+                using (Stream requestStream = await request.GetRequestStreamAsync())
+                {
+                    requestStream.Write(_requestBodyBytes, 0, _requestBodyBytes.Length);
+                }
 
-            using (WebResponse response = await request.GetResponseAsync())
-            using (Stream myStream = response.GetResponseStream())
-            using (var sr = new StreamReader(myStream))
-            {
-                string strContent = sr.ReadToEnd();
-                Assert.True(strContent.Contains(RequestBody));
-            }
+                using (WebResponse response = await request.GetResponseAsync())
+                using (Stream myStream = response.GetResponseStream())
+                using (var sr = new StreamReader(myStream))
+                {
+                    string strContent = sr.ReadToEnd();
+                    Assert.True(strContent.Contains(RequestBody));
+                }
+            }, server => server.HandleRequestAsync(content: RequestBody), options);
         }
 
         [Theory]
-        [MemberData(nameof(EchoServers))]
-        public async Task GetResponseAsync_UseDefaultCredentials_ExpectSuccess(Uri remoteServer)
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task GetResponseAsync_UseDefaultCredentials_ExpectSuccess(bool useSsl)
         {
-            HttpWebRequest request = WebRequest.CreateHttp(remoteServer);
-            request.UseDefaultCredentials = true;
-            var response = await request.GetResponseAsync();
-            response.Dispose();
+            var options = new LoopbackServer.Options { UseSsl = useSsl };
+
+            await LoopbackServer.CreateClientAndServerAsync(async uri =>
+            {
+                HttpWebRequest request = WebRequest.CreateHttp(uri);
+                request.ServerCertificateValidationCallback = (a, b, c, d) => true;
+                request.UseDefaultCredentials = true;
+
+                using WebResponse response = await request.GetResponseAsync();
```

Comment:
Could also be:
```C#
(await request.GetResponseAsync()).Dispose();
```
in which case we wouldn't need an unused variable.

Not a big deal, though.